packages/integration-sdk-runtime/src/synchronization/__tests__/shrinkBatchRawData.test.ts

```diff
@@ -261,4 +261,74 @@ describe('shrinkBatchRawData', () => {
     expect(logger.info).toBeCalledTimes(1);
     expect(logger.info).toHaveBeenCalledWith('Attempting to shrink rawData');
   });
+
+  it('should assign 0 to the size of an undefined property when building the property map', () => {
+    const largeData = new Array(700000).join('aaaaaaaaaa');
+    const data = [
+      {
+        _class: 'test',
+        _key: 'testKey',
+        _type: 'testType',
+        _rawData: [
+          {
+            name: 'test',
+            rawData: {
+              largeRawDataProp: largeData + 'more',
+              testRawData: 'test123',
+              testFinalData: 'test789',
+              anotherLargeRawDataProp: largeData,
+            },
+          },
+        ],
+      },
+      {
+        _class: 'test',
+        _key: 'testKey2',
+        _type: 'testType',
+        largeProperty: largeData,
+        undefProp: undefined,
+        falseyProp: false,
+        _rawData: [
+          {
+            name: 'test2',
+            rawData: {
+              testRawData: 'test123',
+              testFinalData: 'test789',
+            },
+          },
+        ],
+      },
+    ];
+    try {
+      shrinkBatchRawData(data, logger);
```

Comment:
Minor: Can you add an assertion to ensure that this function actually throws. Otherwise, this test could skip all assertions in the below `catch`.

```ts
let shrinkBatchRawDataSucceeded = false;

try {
  shrinkBatchRawData(data, logger);
  shrinkBatchRawDataSucceeded = true;
} catch (err) {

}

expect(shrinkBatchRawDataSucceeded).toEqual(false);
```

Reply:
@austinkelleher Done.